### PR TITLE
Modify pino.final to handle calls without a handler

### PR DESCRIPTION
**Pull request by gyszalai**
Modifies pino.final to be able to return the finalLogger directly. See #490

### CODE REVIEW

File: lib/tools.js [hunk 1]
```diff
@@ -275,14 +275,31 @@ function final (logger, handler) {
   if (typeof logger === 'undefined' || typeof logger.child !== 'function') {
     throw Error('expected a pino logger instance')
   }
-  if (typeof handler !== 'function') {
+  var hasHandler = (typeof handler !== 'undefined')
```
Comment (jsumners) [inline lib/tools.js]: Why `var` instead of `const`?

File: docs/api.md [hunk 2]
```diff
@@ -661,15 +661,12 @@ of each function execution to avoid losing data.
 * See [Reopening log files](/docs/help.md#reopening)
 
 <a id="pino-final"></a>
-### `pino.final(logger, handler) => Function`
 
-The `pino.final` method supplies an exit listener function that can be
-supplied to process exit events such as `exit`, `uncaughtException`,
-`SIGHUP` and so on.
+### `pino.final(logger, [handler]) => Function | Logger`
```
Comment (davidmarkclements) [inline docs/api.md]: can we make it `FinalLogger`

File: lib/tools.js [hunk 3]
```diff
@@ -275,14 +275,31 @@ function final (logger, handler) {
   if (typeof logger === 'undefined' || typeof logger.child !== 'function') {
     throw Error('expected a pino logger instance')
   }
-  if (typeof handler !== 'function') {
+  const hasHandler = (typeof handler !== 'undefined')
+  if (hasHandler && typeof handler !== 'function') {
```
Comment (davidmarkclements) [inline lib/tools.js]: can we update the error message: "if supplied, the handler parameter should be a function"

File: test/final.test.js [hunk 4]
```diff
@@ -147,3 +138,22 @@ test('also instruments custom log methods', async ({pass, fail, error}) => {
   await sleep(10)
   if (passed === false) fail('flushSync not called')
 })
+
+function testLoggers (logger, finalLogger, is, isNot) {
```
Comment (davidmarkclements) [inline test/final.test.js]: I would much prefer that the assertions are repeated inline

for testing repetition > indirection - keep things declarative

File: docs/api.md [hunk 5]
```diff
@@ -680,13 +677,32 @@ is a Node.js stream `pino.final` will throw. It's highly recommended
...
+by any additional arguments the `handler` may be called with.
+
 ```js
 process.on('uncaughtException', pino.final(logger, (err, finalLogger) => {
   finalLogger.error(err, 'uncaughtException')
   process.exit(1)
 }))
 ```
 
+#### `pino.final(logger) => Logger`
```
Comment (davidmarkclements) [inline docs/api.md]: also, FinalLogger.

Comment (gyszalai) [reply docs/api.md]: oops, sorry

**Pull request merged**